internal/auth0/connection/okta/resource.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+
+	"github.com/auth0/terraform-provider-auth0/internal/auth0/connection"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewConnectionResource will return a new auth0_connection_okta resource.
+func NewConnectionResource() *schema.Resource {
+	baseResource := connection.NewBaseConnectionResource(
+		"This resource configure your Okta Enterprise Connection to allow your users to use their enterprise credentials to login to your app.",
+		map[string]*schema.Schema{
+			"client_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The strategy's client ID.",
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "The strategy's client secret.",
+			},
+			"domain": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Domain name.",
+			},
+			"domain_aliases": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Description: "List of the domains that can be authenticated using the identity provider. " +
+					"Only needed for Identifier First authentication flows.",
+			},
+			"scopes": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Description: "Permissions to grant to the connection. Within the Auth0 dashboard these appear " +
+					"under the \"Attributes\" and \"Extended Attributes\" sections. Some examples: " +
+					"`basic_profile`, `ext_profile`, `ext_nested_groups`, etc.",
+			},
+			"issuer": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Issuer URL, e.g. `https://auth.example.com`.",
+			},
+			"jwks_uri": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "JWKS URI.",
+			},
+			"token_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Token endpoint.",
+			},
+			"userinfo_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "User info endpoint.",
+			},
+			"authorization_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Authorization endpoint.",
+			},
+			"icon_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Icon URL.",
+			},
+		},
+		expandConnectionOkta,
+		flattenConnectionOkta,
+	)
+
+	return baseResource
+}
+
+func flattenConnectionOkta(
+	_ *schema.ResourceData,
+	options *management.ConnectionOptionsOkta,
+) (map[string]interface{}, diag.Diagnostics) {
+	m := map[string]interface{}{
+		"strategy":                 "okta",
```

Comment:
Considering we don't have a strategy attribute within the schema now, do we still need to flatten this inside it?